### PR TITLE
read-only mode: api layer

### PR DESCRIPTION
**Pull request by frankban**
Store the user read-only information as an attribute of the API env object.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5341/
Test PASSed.

**Comment by frankban** [comment]
QA:
- `make dist` on this branch: this will generate a `~/code/juju-gui/dist/jujugui-2.1.7.tar.bz2` archive;
- bootstrap a Juju controller after compiling latest master tip: `juju bootstrap lxd lxd --upload-tools --config identity-url=https://api.jujucharms.com/identity`;
- upload the GUI archive to the controller: `juju upgrade-gui ~/code/juju-gui/dist/jujugui-2.1.7.tar.bz2`;
- open the GUI in the browser: `juju gui --show-credentials`, accept the self-signed certificate and the GUI should show up (note that login inputs are kind of weird);
- open the JS console and run the following commands (both must return false): `app.acl.isReadOnly();` and `app.env.get('readOnly');`.

File: jujugui/static/gui/src/app/store/env/api.js [hunk 1]
```diff
@@ -448,21 +448,17 @@ YUI.add('juju-env-api', function(Y) {
       this.pendingLoginResponse = false;
       this.userIsAuthenticated = !data.error;
       if (this.userIsAuthenticated) {
-        // If this is a token login, set the credentials.
         var response = data.response;
-        if (response && response['auth-tag'] && response.password) {
-          this.setCredentials({
-            user: response['auth-tag'],
-            password: response.password
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/store/env/api.js]: Im not sure if this was meant to be removed but I assume so?

**Comment by anthonydillon** [comment]
Code looks good with a probably stupid question in the code :+1:

File: jujugui/static/gui/src/app/store/env/api.js (hunk 1 above)
Comment (frankban) [inline jujugui/static/gui/src/app/store/env/api.js]: Yes, that's a leftover.

**Comment by makyo** [comment]
👍

**Comment by jaycee** [comment]
QA :ok:

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**